google/ggrc-core
# GGRC-4382,3997,4433,3758 Fix restoring a previous state of an instance.

# Issue description

The issue was that on a slow internet connection, when assessment info-pane is opened and the Complete button is clicked before the GET request for the instance is returned, a PUT request for the status change is sent with the empty access_control_list. The issue was introduced in [this commit](https://github.com/google/ggrc-core/pull/6794/commits/9d483cac325cb27428bbef366f1ef854e2700615) as a solution to fix the issue GGRC-3758.

# Steps to test the issue on dev branch

1. Generate or create Assessment
2. On assessment info-pane, move it to "In Progress" state (by changing CA for example)
3. Close info-pane
4. Switch to 'Slow 3G' mode in dev-tools in the Network tab
5. Open this Assessment and quickly click the "Complete" button (before the Assessment GET request is completed)
6. Assessment PUT request will go with the empty access_control_list property.

# Steps to test the changes

1. Perform the steps form the above "fail-test" to check the "access_control_list" is no longer emptied.
2. Test the solution for GGRC-3758:

   1. Invoke Edit assessment modal window
   2. Add person to custom role field
   3. Close the edit window and click 'discard' button
   4. Look at the custom role field on Assessment Info pane. The field should contain the exact number of items as before editing.

3. Check issue with duplicated Audit Captain ( GGRC-4433 )

    1. Open Audit summary page
    2. Open Edit Audit popup
    3. Add at least 3 auditors or Audit captains > Save
    4. Remove 2 auditors/audit captains > Save
    5. Audit captain field should not have duplicated items

# Solution description

1. Revert changes made in the mentioned commit.
2. Make proper solution for GGRC-3758 ( restore the entire instance to its previous state when changes  are discarded )
3. This automatically fixes issues GGRC-4382 and GGRC-3997.

# Sanity checklist

- [ ] __N/A__ ~~My changes are covered by tests.~~

# PR Review checklist
